core/arch/aarch64/clean_call_opt.c

```diff
@@ -156,6 +156,22 @@ instr_is_pop_reg(instr_t *instr, reg_id_t *reg)
     return true;
 }
 
+static inline reg_id_t
+find_nzcv_spill_reg(callee_info_t *ci)
+{
+    int i;
+    reg_id_t spill_reg = DR_REG_INVALID;
+    for (i = NUM_GP_REGS - DR_REG_START_GPR; i >= 0; i--) {
```

Comment:
The initialization of `i` seems slightly odd. You are subtracting an enum value from a number. I think it works out by chance for the test case, because DR_REG_START_GPR is something around 2 on AArch64. But I think it would be better to just subtract 1 from NUM_GP_REGS, to skip over the stack pointer.

Reply:
The intention was to subtract the base (DR_REG_START_GPR) which is 2, thanks for pointing that out. Subtracting 2 is a safer approach.